src/cli.cr

```diff
@@ -76,6 +79,31 @@ module Shards
   end
 
   def self.build(path, args)
+    targets, options = parse_args(args)
+
+    begin
+      Commands::Check.run(path)
+    rescue
+      Commands::Install.run(path)
+    end
+
+    Commands::Build.run(path, targets, options)
+  end
+
+  def self.run_command(path, args)
+    targets, options = parse_args(args)
+
+    begin
+      Commands::Check.run(path)
+    rescue
+      Commands::Install.run(path)
+    end
+
+    Commands::Run.run(path, targets, options)
+  end
+end
```

Comment:
Maybe have a `check_and_install_dependencies` too, remove the `build` and `run_command` methods and inline the 3 calls in  the case above:

```crystal
targets, options = parse_args(args)
chech_and_install_dependencies
Command::Run.run(path, targets, options)
```